libs/wallet/src/connect-dialog/json-rpc-connector-form.tsx

```diff
@@ -0,0 +1,252 @@
+        );
+      }
+
+      // Start connection flow. User will be prompted to select a wallet and enter
+      // its password in the wallet application, promise will resolve once successful
+      // or it will throw
+      setStatus(Status.connecting);
+      await connector.connectWallet();
+
+      // Check wallet is permitted to reveal its public keys
+      setStatus(Status.gettingPerms);
+      const permsResult = await connector.getPermissions();
+      if (permsResult.permissions.public_keys === 'none') {
+        // Automatically request new perms. User will again be prompted to permit this change
+        // and enter their password
+        setStatus(Status.requestingPerms);
+        await connector.requestPermissions();
+      }
+
+      // Call connect in the wallet provider. The connector will be stored for
+      // future actions such as sending transactions
+      await connect(connector);
+
+      setStatus(Status.connected);
+
+      // Briefly allow the success state to be shown then automatically
+      // close the dialog
+      setTimeout(() => {
+        onConnect();
+      }, CLOSE_DELAY);
+    } catch (err) {
+      if (err instanceof WalletError) {
+        setError(err);
+      }
+      setStatus(Status.error);
+    }
+  }, [connector, connect, onConnect, walletUrl, appChainId]);
+
+  useEffect(() => {
+    if (status === Status.idle) {
+      attemptConnect();
+    }
+  }, [status, attemptConnect]);
+
+  if (status === Status.idle) {
+    return null;
+  }
+
+  return <Connecting status={status} error={error} connector={connector} />;
+};
+
+const Connecting = ({
+  status,
+  error,
+  connector,
+}: {
+  status: Status;
+  error: WalletError | null;
+  connector: JsonRpcConnector;
+}) => {
+  if (status === Status.error) {
+    let title = t('Something went wrong');
+    let text: ReactNode | undefined = t('An unknown error occurred');
+
+    if (error) {
+      if (error.code === ClientErrors.NO_SERVICE.code) {
+        title = t('No wallet detected');
+        text = t(`No wallet application running at ${connector.url}`);
+      } else if (error.code === ServiceErrors.CONNECTION_DECLINED) {
+        title = t('Connection declined');
+        text = t('Your wallet connection was rejected');
+      } else if (error.code === ServiceErrors.NO_HEALTHY_NODE) {
+        title = error.message;
+        text = (
+          <>
+            {capitalize(error.data)}
+            {'. '}
+            <Link href="https://docs.vega.xyz/docs/mainnet/concepts/vega-wallet">
+              {t('Read the docs to troubleshoot')}
+            </Link>
+          </>
+        );
+      } else if (error.code === 0) {
+        title = t('Wrong network');
+        text = (
+          <>
+            {t(`To complete your wallet connection, set your wallet network in your
+            app to ${'TODO: App chain id'} then `)}
```

Comment:
This will be addressed by https://github.com/vegaprotocol/frontend-monorepo/issues/714

Reply:
This TODO was actually accidentally not complete by me. In this case the incorrect chain message here is for the Vega network, not Ethereum

![Screen Shot 2022-09-27 at 8 37 59 PM](https://user-images.githubusercontent.com/6803987/192682020-8ebaa5db-f997-4cb8-a77f-f99597792e12.jpg)